### PR TITLE
Click style prompts and choices

### LINKED ISSUE #441: Prompt list of choices if list in cookiecutter.json
Example:

    {
        "status_classifier": [
            "Development Status :: 1 - Planning",
            "Development Status :: 2 - Pre-Alpha",
            "Development Status :: 3 - Alpha",
            "Development Status :: 4 - Beta",
            "Development Status :: 5 - Production/Stable",
            "Development Status :: 6 - Mature",
            "Development Status :: 7 - Inactive"
        ],
    }

... would be prompted like

    Select status_classifier:
        1 - Development Status :: 1 - Planning
        2 - Development Status :: 2 - Pre-Alpha
        3 - Development Status :: 3 - Alpha
        4 - Development Status :: 4 - Beta
        5 - Development Status :: 5 - Production/Stable
        6 - Development Status :: 6 - Mature
        7 - Development Status :: 7 - Inactive
    Enter your choice (1, 2, 3, 4, 5, 6 or 7):

### PR DESCRIPTION
**Pull request by hackebrot**
This PR introduces new functions for prompting the user for input. I decided to let [click](https://github.com/mitsuhiko/click) handle all the compatibility matters and go with its prompt display as I like it better than what we used to have. :smiley: 

Additionally this PR also provides support for list variables (resolves #441).

Please thoroughly review my changes as I touched some core functions and I want to be certain I did not break anything. :exclamation: Feel free to use https://github.com/hackebrot/cookiecutter-kivy/commit/58d2aa91e819508e99f385316e679a26411c883e for manual testing of the new prompts as well as choices support.

Very much looking forward to your feedback. :bow: 

```bash
(cookiecutter)[raph@imac hackebrot]$ cookiecutter cookiecutter-kivy/
full_name [Raphael Pierzina]: Testing Goat
github_username [hackebrot]: 
project_name [Kivy Project]: Foo Bar      
repo_name [foobar]: 
project_short_description [Cookiecutter template kivy apps.]: Testing Choices
release_date [2014-05-15]: 2015-05-25 
year [2014]: 2015
version [0.1.0]: 
minimum_kivy_version [1.8.0]: 1.9.0
Select orientation:
1 - all
2 - landscape
3 - portrait
Choose from 1, 2, 3 [1]: 2
(cookiecutter)[raph@imac hackebrot]$ 
```

### CODE REVIEW

File: cookiecutter/prompt.py [hunk 1]
```diff
@@ -17,21 +17,88 @@
...
+    :return: Exactly one item of ``options`` that has been chosen by the user
+    """
+    # Please see http://click.pocoo.org/4/api/#click.prompt
+    if not isinstance(options, list):
+        raise TypeError
+
+    if not options:
+        raise ValueError
+
+    if not all(is_string(opt) for opt in options):
```
Comment (pydanny) [inline cookiecutter/prompt.py]: This line needs to be documented.

File: cookiecutter/prompt.py [hunk 2]
```diff
@@ -17,21 +17,88 @@
...
+    ))
+
+    user_choice = click.prompt(
+        prompt, type=click.Choice(choices), default=default
+    )
+    return choice_map[user_choice]
+
+
+def _render_variable(env, raw, cookiecutter_dict):
+    raw = raw if is_string(raw) else str(raw)
```
Comment (pydanny) [inline cookiecutter/prompt.py]: I'm not a fan of ternary operators as they are not as clear as breaking it up into multiple lines of code.

File: cookiecutter/prompt.py [hunk 3]
```diff
@@ -49,55 +116,17 @@ def prompt_for_config(context, no_input=False):
...
+                cookiecutter_dict, env, key, raw, no_input
+            )
+        else:
+            # We are dealing with a regular variable
+            val = _render_variable(env, raw, cookiecutter_dict)
 
-            if new_val != '':
-                val = new_val
+            if not no_input:
+                val = read_user_variable(key, val)
```
Comment (pydanny) [inline cookiecutter/prompt.py]: I like this change as it is much more clear in what it is doing. :smile:

File: cookiecutter/prompt.py (hunk 1 above)
Comment (hackebrot) [inline cookiecutter/prompt.py]: This is actually a mistake on my part. `click` requires the actual choices to be strings however I am not using them directly but generate numbers on the fly.

Removed this part as well as the corresponding tests.

:bow:

File: cookiecutter/prompt.py (hunk 2 above)
Comment (hackebrot) [inline cookiecutter/prompt.py]: Changed this one and implemented tests for both cases. See 9ccb1ec2ee

**Pull request merged**